models/git/branch.go

```diff
@@ -166,6 +166,11 @@ func GetBranch(ctx context.Context, repoID int64, branchName string) (*Branch, e
 			RepoID:     repoID,
 			BranchName: branchName,
 		}
+	} else if branch.IsDeleted {
+		return nil, ErrBranchNotExist{
+			RepoID:     repoID,
+			BranchName: branchName,
+		}
```

Comment:
@lunny it seems to be a longstanding bug.

Reply:
It should be checked out of the function because sometimes we need to get the deleted branch. I can see most places have checked `.IsDeleted` after invoking this function. I will check all of them and send a pull request.

Reply:
It is quite misleading, and I spent hours on debugging the problem:

I do not think everyone could clearly remember to check `IsDeleted` everytime everywhere.

![image](https://github.com/user-attachments/assets/f2ac1788-e762-4675-a094-529f865d9b8c)

Reply:
I reverted the GetBranch behavior in 3ec97d05d6154c59ba06c61f402c2ff6be45a05a